src/devices/tuya.ts

```diff
@@ -811,6 +811,18 @@ const definitions: DefinitionWithExtend[] = [
         toZigbee: [],
         exposes: [e.gas(), e.tamper()],
     },
+    {
+        fingerprint: [
+            {
+                modelID: 'TS0205',
+                manufacturerName: '_TZ3210_up3pngle',
+            },
+        ],
+        model: 'TS0205_smoke_2',
+        vendor: 'Tuya',
+        description: 'Smoke sensor',
+        extend: [battery(), iasZoneAlarm({zoneType: 'smoke', zoneAttributes: ['alarm_1', 'tamper']})],
```

Comment:
Does the battery % work? It's not working for `TS0205`

Reply:
Yes it does 😃

Reply:
Could you try with `battery({percentageReporting: false})` , re-pair the device and see if it still works? If yes, then you can apply these changes to the existing `TS0205` definition.

Reply:
First results  with ```battery({percentageReporting: false})```, after 5 hours:
Sensor is not reporting battery level by itself at moment of re-pair and has still not reported it after 5 hours.
![image](https://github.com/user-attachments/assets/2bd05fad-5481-4a95-a643-a30b3ca0b338)


(test on second sensor with this setting: after awakening this sensor ( trigger tamper by opening), and hitting the blue (![image](https://github.com/user-attachments/assets/aa47be01-ee27-4ba0-89a8-45af599259cb)  )button, the battery % is shown. So it look like it does not work 100% with this ```battery({percentageReporting: false})``` setting. 

Now waiting for the first sensor to show tell it is still alive. (Normally the sensors did that after aprox. every 18 hours with uploaded settings)

Further results tomorrow ...

Reply:
So now, 29 hours after the re-pair, the device has not send anything; battery is still None/? and last-seen is still the same ...
I guess this shows that the ```battery({percentageReporting: false})``` does not work with the ```_TZ3210_up3pngle``` device. I checked the settings to be sure that all OK, and they are!

Last test: I woke up the device by opening it, triggering the tamper: ```last-seen``` is updated and battery line is added to the status (state.yaml) but value is still ```null```. After that, I woke up the device again and hit the (![image](https://github.com/user-attachments/assets/aa47be01-ee27-4ba0-89a8-45af599259cb)  )button: battery-% appears (94%).

(PS: the other 4 devices, interviewed with my proposed settings wereworking OK and did send new battery % ans last-seen info during test-time.)

Reply:
Then I would propose to use the following definition:

```
{
        zigbeeModel: ['TS0205'],
        model: 'TS0205',
        vendor: 'Tuya',
        description: 'Smoke sensor',
        whiteLabel: [
            {vendor: 'Tesla Smart', model: 'TSL-SEN-SMOKE'},
            {vendor: 'Dongguan Daying Electornics Technology', model: 'YG400A'},
            tuya.whitelabel('Tuya', 'TS0205_smoke_2', 'Smoke sensor', ['_TZ3210_up3pngle']),
        ],
        // Configure battery % fails
        // https://github.com/Koenkk/zigbee2mqtt/issues/22421
        extend: [battery({percentageReporting: false}), iasZoneAlarm({zoneType: 'smoke', zoneAttributes: ['alarm_1', 'tamper']})],
        configure: async (device, coordinatorEndpoint) => {
            if (device?.manufacturerName === '_TZ3210_up3pngle') {
                // Required for this version
                // https://github.com/Koenkk/zigbee-herdsman-converters/pull/8004
                const endpoint = device.getEndpoint(1);
                await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
                await reporting.batteryPercentageRemaining(endpoint);
            }
        },
}
```

Reply:
So simplify by modernExtent is not a priority?